File: node/sol_02.js
function xorBuffers(buf1, buf2) {
  if (buf1.length !== buf2.length) {
    throw new Error("Buffers must be of equal length");
  }

  const result = Buffer.alloc(buf1.length);

  for (let i = 0; i < buf1.length; i++) {
    result[i] = buf1[i] ^ buf2[i];
  }

  return result;
}

const buf1 = Buffer.from("1c0111001f010100061a024b53535009181c", "hex");
const buf2 = Buffer.from("686974207468652062756c6c277320657965", "hex");

const xorResult = xorBuffers(buf1, buf2);
console.log(xorResult.toString("hex"));

File: go/sol_02.go
package main

import (
	"encoding/hex"
	"fmt"
)

func xorBuffers(buf1, buf2 []byte) []byte {
	if len(buf1) != len(buf2) {
		panic("Buffers must be of equal length")
	}

	result := make([]byte, len(buf1))

	for i := 0; i < len(buf1); i++ {
		result[i] = buf1[i] ^ buf2[i]
	}

	return result
}

func main() {
	buf1, _ := hex.DecodeString("1c0111001f010100061a024b53535009181c")
	buf2, _ := hex.DecodeString("686974207468652062756c6c277320657965")

	xorResult := xorBuffers(buf1, buf2)
	fmt.Println(hex.EncodeToString(xorResult))
}
